core/src/main/java/io/questdb/log/TemplateParser.java

```diff
@@ -0,0 +1,244 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.log;
+
+import io.questdb.std.*;
+import io.questdb.std.datetime.DateFormat;
+import io.questdb.std.datetime.microtime.TimestampFormatCompiler;
+import io.questdb.std.datetime.microtime.TimestampFormatUtils;
+import io.questdb.std.str.CharSink;
+import io.questdb.std.str.StringSink;
+
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class TemplateParser implements Sinkable {
+
+    public static CharSequenceObjHashMap<CharSequence> adaptMap(Map<String, String> props) {
+        CharSequenceObjHashMap<CharSequence> properties = new CharSequenceObjHashMap<>(props.size());
+        for (String key : props.keySet()) {
+            properties.put(key, props.get(key));
+        }
+        return properties;
+    }
+
+    private static final String DATE_FORMAT_KEY = "date:";
+    private static final int NIL = -1;
+
+    private final TimestampFormatCompiler dateCompiler = new TimestampFormatCompiler();
+    private final StringSink resolveSink = new StringSink();
+    private final ObjList<Component> txtComponents = new ObjList<>();
```

Comment:
inconsistent naming of variables:

<img width="707" alt="image" src="https://user-images.githubusercontent.com/7276403/144147448-c8a2dc3f-c350-4ac9-be66-a56408b0c592.png">